src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -354,10 +354,44 @@ class CodeBlock private constructor(
      *     Shouldn't contain braces or newline characters.
      */
     fun beginControlFlow(controlFlow: String, vararg args: Any?) = apply {
-      add(controlFlow + " {\n", *args)
+      add("$controlFlow {\n", *args)
       indent()
     }
 
+    /**
+     * @param controlFlow the control flow construct and its code, such as "if (foo == 5)".
+     *     Shouldn't contain braces or newline characters.
+     * @param parameterNames optional control flow parameter names. Allow producing expressions such
+     * as:
+     *
+     * ```
+     * list.forEachIndexed { index, value ->
+     * ```
+     */
+    fun beginParameterizedControlFlow(controlFlow: String, vararg parameterNames: String) =
```

Comment:
This seems like a weird name to describe what always seems to be a lambda. Is it ever not a lambda? I guess you could argue that it's control flow-like, but it seems like a stretch of the term. And then my other concern is that "parameterized" is a term that's usually associated with generics.

Reply:
Good point, the naming is off here. I can try changing it to `beginLambdaExpression` and tinker with param names to see what works better.